enzyme/Enzyme/Enzyme.cpp

```diff
@@ -3658,10 +3658,12 @@ void augmentPassBuilder(llvm::PassBuilder &PB) {
   PB.registerFullLinkTimeOptimizationEarlyEPCallback(loadLTO);
 }
 
-extern "C" void registerEnzyme(llvm::PassBuilder &PB) {
-#ifdef ENZYME_RUNPASS
-  augmentPassBuilder(PB);
-#endif
+// A version used by Rust-Enzyme, which does not require cmake adjustments.
+// Rust just always sets augment to true.
+extern "C" void registerEnzyme2(llvm::PassBuilder &PB, bool augment = false) {
```

Comment:
can we come up with a better name, like registerEnzymeAndPassPipeline

and we shouldn't need two versions of this if we have the optional arg?

Reply:
Yep, that was my question:
> So my confusion comes from your recommendation to handle Rust and C++ different here.

and 

> Doing it at runtime would be totally fine for me, but then I wonder why you don't do the same with the clang/lld plugin? 

Shall I then remove ENZYME_RUNPASS completely and update the clang/lld usages to set it to true?

Reply:
We can not I guess, seems like the registration process has a fixed interface?
Feel free to push your own changes and merge it.

```
/home/manuel/prog/Enzyme/enzyme/Enzyme/Enzyme.cpp:3704:59: error: cannot initialize a member subobject of type 'void (*)(PassBuilder &)' with an lvalue of type 'void (llvm::PassBuilder &, bool)': different number of parameters (1 vs 2)
 3704 |   return {LLVM_PLUGIN_API_VERSION, "EnzymeNewPM", "v0.1", registerEnzymeAndPassPipeline};
      |                                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
okay in that case just give the functions more reasonable names then registerEnzyme and registerEnzymeAndPassPipeline (being the false / true version), one calling the other

Reply:
done